idx/memory/memory.go

```diff
@@ -376,76 +376,161 @@ func (m *MemoryIdx) GetPath(orgId int, path string) []idx.Archive {
 	return archives
 }
 
-func (m *MemoryIdx) Tag(orgId int, tag string, from int64) map[string]uint32 {
+func (m *MemoryIdx) TagDetails(orgId int, key, filter string, from int64) (map[string]uint64, error) {
 	if !tagSupport {
 		log.Warn("memory-idx: received tag query, but tag support is disabled")
-		return nil
+		return nil, nil
+	}
+
+	var re *regexp.Regexp
+	if len(filter) > 0 {
+		if filter[0] != byte('^') {
+			filter = "^(?:" + filter + ")"
+		}
+		var err error
+		re, err = regexp.Compile(filter)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	m.RLock()
 	defer m.RUnlock()
 
-	var tags TagIndex
-	var ok bool
-	if tags, ok = m.Tags[orgId]; !ok {
-		return nil
+	tree, ok := m.tags[orgId]
```

Comment:
because we have:
```
type MemoryIdx struct {
	sync.RWMutex
	DefById map[string]*idx.Archive
	Tree    map[int]*Tree
	tags    map[int]TagIndex
}
```
let's stick to the name tag (or tagIndex) when we get a substructure from tags, and the name tree when we get a substructure from Tree

Reply:
done https://github.com/grafana/metrictank/pull/750/commits/5118967a09b4c63b060f82eb487ad2d335eb8e74